impl/src/test/java/io/kroxylicious/testing/kafka/invm/LoggingPrintStreamTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.testing.kafka.invm;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@ExtendWith(MockitoExtension.class)
+class LoggingPrintStreamTest {
+
+    @Mock
+    private System.Logger logger;
+
+    @Test
+    void singleLog() {
+        try (var lps = LoggingPrintStream.loggingPrintStream(logger, System.Logger.Level.INFO)) {
+            lps.println("Hello, world");
+
+            verify(logger, times(1)).log(System.Logger.Level.INFO, "{0}", "Hello, world");
+        }
+        verifyNoMoreInteractions(logger);
```

Comment:
Given every single test case calls `verifyNoMoreInteractions(logger)` I wonder if it should really be pulled out to `afterEach`? 

Equally the warning from the mockito docs about [`verifyNoMoreInteractions`](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#finding_redundant_invocations) seems pertinent 
> A word of warning: Some users who did a lot of classic, expect-run-verify mocking tend to use verifyNoMoreInteractions() very often, even in every test method. verifyNoMoreInteractions() is not recommended to use in every test method. verifyNoMoreInteractions() is a handy assertion from the interaction testing toolkit. Use it only when it's relevant. Abusing it leads to overspecified, less maintainable tests.

So I'm kinda on the fence about it.

Reply:
My intent was to ensure that the class wasn't say, double logging, on close() or flush().  If I remove `verifyNoMoreInteractions`, then the test doesn't detect it.  I'll move verifyNoMoreInteractions to afterEach().  Unless there is a better way?

Reply:
Nope I don't any better way!